editor/src/components/custom-code/code-file.ts

```diff
@@ -254,3 +273,169 @@ export function codeCacheToBuildResult(cache: {
+          )
+        } else {
+          // Now need to identify the element relating to the last part of the dropped scene path.
+          const lastScenePathPart = droppedPathPart[droppedPathPart.length - 1]
+
+          // Walk the parsed representation to find the element with the given uid.
+          const parsedContent = currentFile.fileContents.parsed
+          let targetElement: JSXElement | null = null
+          for (const topLevelElement of parsedContent.topLevelElements) {
+            const possibleTarget = findElementWithUID(topLevelElement, lastScenePathPart)
+            if (possibleTarget != null) {
+              targetElement = possibleTarget
+              break
+            }
+          }
+
+          // Identify where the component is imported from or if it's in the same file.
+          if (targetElement == null) {
+            return normalisePathImportNotFound()
+          } else {
+            const nonNullTargetElement: JSXElement = targetElement
+            function lookupElementImport(elementBaseVariable: string): NormalisePathResult {
+              const importedFrom = importedFromWhere(elementBaseVariable, parsedContent.imports)
+              if (importedFrom == null) {
+                return normalisePathImportNotFound()
+              } else {
+                if (importedFrom === 'utopia-api' && elementBaseVariable === 'Scene') {
+                  // Navigate around the scene with the special case handling.
+                  const componentAttr = getJSXAttribute(nonNullTargetElement.props, 'component')
+                  if (componentAttr != null && isJSXAttributeOtherJavaScript(componentAttr)) {
+                    return lookupElementImport(componentAttr.javascript)
+                  } else {
+                    return normalisePathError(
+                      `Unable to handle Scene component definition for ${TP.toString(elementPath)}`,
+                    )
+                  }
+                } else if (importedFrom.includes('/')) {
+                  const absoluteImportedPath = absolutePathFromRelativePath(
+                    currentFilePath,
+                    importedFrom,
+                  )
+                  return normalisePathToUnderlyingTarget(
+                    projectContents,
+                    absoluteImportedPath,
+                    potentiallyDroppedFirstSceneElementResult.newPath,
+                  )
+                } else {
+                  return normalisePathEndsAtDependency(importedFrom)
+                }
+              }
+            }
+            // Handle things like divs.
+            if (isIntrinsicElement(targetElement.name)) {
+              return normalisePathSuccess(
+                TP.dynamicPathToStaticPath(potentiallyDroppedFirstSceneElementResult.newPath),
+                currentFilePath,
+                currentFile,
+              )
+            } else {
+              return lookupElementImport(targetElement.name.baseVariable)
```

Comment:
This needs to be extended to support the fully qualified element name (I think this would break right now on `animated.div`)

Reply:
Oh sorry, I'm an idiot. I just realised what `baseVariable` is :facepalm:

Reply:
Shouldn't do, as `animated` is the value in `baseVariable`, which would be the reference to find in the imports.